stochatreat/stochatreat.py

```diff
@@ -215,11 +215,8 @@ def stochatreat(data: pd.DataFrame,
     ids_treats = ids_treats.sort_values(by=idx_col)
     # map the concatenated blocks to block ids to retrieve the blocks
     # within which randomization was done easily
-    concatenated_blocks_unique = ids_treats["block"].unique()
-    n_unique = len(concatenated_blocks_unique)
-    concatenated_blocks_map = dict(zip(concatenated_blocks_unique, range(n_unique)))
-    ids_treats["block_id"] = ids_treats["block"].map(concatenated_blocks_map)
-    ids_treats = ids_treats.drop(columns="block")
+    ids_treats["block"] = ids_treats.groupby(["block"]).ngroup()
+    ids_treats.rename({"block": "block_id"}, axis="columns", inplace=True)
```

Comment:
pedantry: As I've learned, [`inplace` is discouraged](https://tomaugspurger.github.io/method-chaining.html) and I find `rename(columns={"block": "block_id"})` a little easier to read.

Reply:
good to know, thanks!

Reply:
On second look: why not assign to `block_id` right away instead of renaming?

Reply:
@redouane-dziri Can you address this and then squash all commits into a single one? (https://github.com/wprig/wprig/wiki/How-to-squash-commits)

Reply:
I renamed because i squashed the old "block" column (with concatenated string values of the block columns), would you rather I assign a new column and then delete "block"?

Reply:
Assigning to "block_id" leaves the old column untouched and you save yourself the rename, no?

Reply:
Yes but I don't want to keep the old column, so I still need to remove it after (no rename but one drop) - this way is clearer anyway, with the renaming it can be confusing